language/tools/test_generation/src/control_flow_graph.rs

```diff
@@ -266,20 +266,52 @@ impl CFG {
                 basic_block.locals_in =
                     cfg_copy.merge_locals(cfg_copy.get_parent_ids(block_id as BlockIDSize));
             }
-            // basic_block.locals_out = basic_block.locals_in.clone();
             basic_block.locals_out = CFG::vary_locals(&mut rng, basic_block.locals_in.clone());
         }
     }
 
+    /// Decide the serialization order of the blocks in the CFG
+    pub fn serialize_block_order(&self) -> Vec<BlockIDSize> {
+        let mut block_order: Vec<BlockIDSize> = Vec::new();
+        let mut block_queue: VecDeque<BlockIDSize> = VecDeque::new();
+        block_queue.push_back(0);
+        while !block_queue.is_empty() {
+            let block_id_front = block_queue.pop_front();
+            // The queue is non-empty so the front block id will not be none
+            assume!(block_id_front.is_some());
+            let block_id = block_id_front.unwrap();
+            let child_ids = self.get_children_ids(block_id);
+            if child_ids.len() == 2 {
+                block_queue.push_front(child_ids[0]);
+                block_queue.push_back(child_ids[1]);
+            } else if child_ids.len() == 1 {
+                block_queue.push_back(child_ids[0]);
+            } else if child_ids.len() != 0 {
+                unreachable!(
+                    "Invalid number of children for basic block {:?}",
+                    child_ids.len()
+                );
+            }
+            if !block_order.contains(&block_id) {
```

Comment:
This operation can be expensive. Perhaps add a comment to explain why this is not expected to be a problem.